src/arch/amdgpu/vega/operand.hh

```diff
@@ -514,6 +514,29 @@ namespace VegaISA
         {
             assert(NumDwords == 1 || NumDwords == 2);
 
+            if (_opIdx >= REG_INT_CONST_POS_MIN &&
```

Comment:
Please just add a comment explaining what this is doing -- I think I get it, but not 100% sure.

Reply:
I will add a comment saying something along the lines of "there are ~80 inline constants" so using a switch statement is painful.  The second check I put in was too avoid too much code duplication. 

On a more technical note, it seems GCC 13's static analyzer is testing for all possible values of this `_opIdx` even though this method would never be called for values not checked here (panic in the default just in case).  I think making `isConstVal` constexpr would work, but I was concerned that would break older compiler versions that are still (for now) supported officially by gem5

Reply:
Sounds great, thanks!